test/framework/container_engine.go

```diff
@@ -103,7 +103,13 @@ func (containerEngine *containerEngineStruct) PushImage(imageTag string) Contain
 
 func (containerEngine *containerEngineStruct) BuildImage(projectLocation, imageTag string) ContainerEngine {
 	if containerEngine.err == nil {
-		_, containerEngine.err = CreateCommand(containerEngine.engine, containerEngine.buildCommand, "--tag", imageTag, ".").InDirectory(projectLocation).WithLoggerContext(containerEngine.namespace).Execute()
+		retries := 3
+		for i := 0; i < retries; i++ {
+			_, containerEngine.err = CreateCommand(containerEngine.engine, containerEngine.buildCommand, "--tag", imageTag, ".").InDirectory(projectLocation).WithLoggerContext(containerEngine.namespace).Execute()
+			if containerEngine.err == nil {
+				return containerEngine
+			}
+		}
```

Comment:
If this is a long term solution, I would implement the "Retry" function as part of the CreateCommand framework (see line [here](https://github.com/kiegroup/kogito-cloud-operator/blob/3613717d8111cb5f696f9235ee3e5501aebea3f9/test/framework/command.go#L67)). This way we can reuse this feature in other commands in the future. The usage would be simply:

```java
CreateCommand(...).InDirectory(...).WithRetry(3 times)
```

What do you think?
```

Reply:
hmm, good point.
will extend the command executor.